plutus-core/testlib/Test/Tasty/Extras.hs

```diff
@@ -21,32 +24,47 @@ import Data.ByteString.Lazy qualified as BSL
 import Data.Text (Text)
 import Data.Text.Encoding (encodeUtf8)
 import Data.Version
-import System.FilePath ((</>))
+import System.FilePath (joinPath, (</>))
 import System.Info
 import Test.Tasty
 import Test.Tasty.Golden
 
+ghcVersion :: String
+ghcVersion = showVersion compilerVersion
+
+-- Given two lists of FilePaths, concatenate the members of the first list
+-- together, append the GHC version number, then append the concatenation of the
+-- members of the second list.
+makeVersionedFilePath :: [FilePath] -> [FilePath] -> FilePath
+makeVersionedFilePath left right = joinPath left </> ghcVersion </> joinPath right
+
 -- | A 'TestTree' of tests under some name prefix.
-type TestNested = Reader [String] TestTree
+type TestNested = Reader [FilePath] TestTree
 
 -- | Run a 'TestTree' of tests with a given name prefix.
-runTestNestedIn :: [String] -> TestNested -> TestTree
+runTestNestedIn :: [FilePath] -> TestNested -> TestTree
 runTestNestedIn path test = runReader test path
 
 -- | Run a 'TestTree' of tests with an empty prefix.
 runTestNested :: TestNested -> TestTree
 runTestNested = runTestNestedIn []
 
 -- | Descend into a name prefix.
-testNested :: String -> [TestNested] -> TestNested
+testNested :: FilePath -> [TestNested] -> TestNested
 testNested folderName =
     local (++ [folderName]) . fmap (testGroup folderName) . sequence
 
 -- | Like `testNested` but adds a subdirectory corresponding to the GHC version being used.
-testNestedGhc :: String -> [TestNested] -> TestNested
-testNestedGhc folderName = testNested (folderName </> ghcVer)
-  where
-    ghcVer = showVersion compilerVersion
+testNestedGhc :: FilePath -> [TestNested] -> TestNested
+testNestedGhc folderName = testNested (folderName </> ghcVersion)
+
+-- Create a TestTree which runs in the directory 'path/<ghc-version>/dir'
+runTestGroupNestedGhcIn :: [FilePath] -> FilePath -> [TestNested] -> TestTree
```

Comment:
I don't like this name very much, but I couldn't think of anything better.

Reply:
I'm not sure why this is needed. There's doesn't seem to be a use case where `dir` isn't `[]`?

Reply:
Oh, that's wrong.  It says it puts it in `path/version/dir`, but in fact it puts it in `path/dir/version`. 

This gets used in `marlowe/test` and `lists/test`.  See [here](https://github.com/input-output-hk/plutus/blob/74860632463e08989a5504e75477163e6b07125c/plutus-benchmark/marlowe/test/Spec.hs#L48) for example, where the directory is `semantics`.  You end up with golden files in `plutus-benchmark/marlowe/test` in the following directories:

* role-payout/9.2
* role-payout/9.6
* semantics/9.2
* semantics/9.6

What I meant to do in that function was to move the version directories further up, so you'd have

* 9.2/role-payout
* 9.2/semantics
* 9.6/role-payout
* 9.6/semantics

I don't know if that's any better.  Maybe we should just leave the directory layout as it is and get rid of the final `dir` argument in that function.

Reply:
I've fixed this now: all of the `9.2`/`9.6`/etc directories should be at the leaves of the directory hierarchy now.